Source/LinqToDB/Linq/Builder/ExpressionBuilder.SqlBuilder.cs

```diff
@@ -185,7 +185,7 @@ public bool IsSupportedSubquery(IBuildContext parent, IBuildContext context, out
 				SqlJoinedTable? fakeJoin = null;
 
 				// add fake join there is no still reference
-				if (null == clonedParentContext.SelectQuery.Find(e => e is SelectQuery sc && sc == clonedContext.SelectQuery))
+				if (clonedParentContext.SelectQuery.From.Tables.Count > 0 && null == clonedParentContext.SelectQuery.Find(e => e is SelectQuery sc && sc == clonedContext.SelectQuery))
```

Comment:
Awfully long - can we break this up over multiple lines?
```suggestion
				if (clonedParentContext.SelectQuery.From.Tables.Count > 0 
					&& clonedParentContext.SelectQuery.Find(e => e is SelectQuery sc && sc == clonedContext.SelectQuery) == null)
```